nabs/preprocessors.py

```diff
@@ -0,0 +1,291 @@
+"""
+Wrappers and decorators to modify existing plans.
+
+This is the LCLS counterpart to bluesky.preprocessors.
+
+This module contains "wrapper" functions that take a plan as an argument
+and yield messages from a new, modified plan, as well as "decorator"
+functions that can be applied to bluesky plan functions to return new
+plan functions with modifications.
+"""
+from functools import wraps
+
+import bluesky.plan_stubs as bps
+import bluesky.preprocessors as bpp
+from bluesky.utils import make_decorator
+
+
+def _get_daq():
+    """
+    Helper function to get the active DAQ object.
+
+    This also wraps the pcdsdaq import because pcdsdaq is an optional
+    dependency of nabs. This will fail unless pcdsdaq is installed.
+
+    Returns
+    -------
+    daq: Daq
+        The DAQ (data aquisition) system bluesky-compatible control object.
+    """
+
+    from pcdsdaq.daq import get_daq  # NOQA
+    return get_daq()
+
+
+class _Dummy:
+    """
+    Class to sub in for the DAQ when we need to drop a message.
+
+    You cant just remove a message entirely with plan_mutator, you need
```

Comment:
I think this is the only way to do it without causing potential issues- I have to apply a stage decorator to make sure the daq works properly, but I have to prevent stage/unstage from being run twice, but I can't just remove a message entirely.